src/aioprometheus/asgi/middleware.py

```diff
@@ -148,7 +148,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send):
             await self.asgi_callable(scope, receive, send)
             return
 
-        if scope["type"] == "http":
+        if scope["type"] in ("http", "websocket"):
```

Comment:
Nice, this almost fixed it for me 🤩. At least with starlette/fastapi it just needed a small additional change, as this now crashes in line 174 for me, because websockets do not have a `method` property. This can be fixed by either adding `websocket` to the scope check in line 147 (and then not having metrics for the websocket connections) or by replacing line 174 with

```
        method = scope.get("method")
```

Reply:
Oh yes, that's right. I've already patched that in my project too. But I haven't transferred it here yet. I'll see if I can do that in the course of the day.

Reply:
I also have a branch with that already, so if you’re fine with that, I could open a PR and save you the work.

Reply:
oh, i already pushed my changes :see_no_evil:

Reply:
ah nice, your solution is much better